homeassistant/components/tcp/binary_sensor.py

```diff
@@ -1,22 +1,33 @@
 """Provides a binary sensor which gets its values from a TCP socket."""
+from __future__ import annotations
+
+from typing import Any, Final
+
 from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType
 
-from .sensor import CONF_VALUE_ON, PLATFORM_SCHEMA, TcpSensor
+from .const import CONF_VALUE_ON
+from .sensor import PLATFORM_SCHEMA as TCP_PLATFORM_SCHEMA, TcpSensor
```

Comment:
A future improvement would be to refactor the placement of the schema and extract a common base class for the TCP entities. A platform should not import anything from another platform.

Reply:
Many thanks for this proposal :+1: